heap/heap.py

```diff
@@ -22,21 +26,43 @@ def get_size(self):
   def _bubble_up(self, index):
     child = self.storage[index]
     parent = self.storage[floor((index - 1) / 2)]
+    #print(f'{self.storage} parent: {parent}, child: {child}')
     #If parent is less than child, swap
-    if parent < child:
+    while parent < child:
       self.storage[floor((index - 1) / 2)], self.storage[index] = self.storage[index], self.storage[floor((index - 1) / 2)]
+      index = floor((index - 1) / 2)
+      if index > 0:
+        parent = self.storage[floor((index - 1) / 2)]
+      else:
+        break
+
+  def get_left_child(self, index):
+    n = (2 * index) + 1
+    if n < len(self.storage):
+      return self.storage[n]
+    return float("-inf")
+
+  def get_right_child(self, index):
+    n = (2 * index) + 2
+    if n < len(self.storage):
+      return self.storage[n]
+    return float("-inf")
 
   def _sift_down(self, index):
-    p = (index-1)/2
-    print(p)
-
-
-heap = Heap()
-heap.insert(6)
-heap.insert(8)
-heap.insert(10)
-heap.insert(9)
-heap.insert(1)
-heap.insert(9)
-heap.insert(9)
-heap.insert(5)
\ No newline at end of file
+    current_node = self.storage[index]
+    left = self.get_left_child(index)
+    right = self.get_right_child(index)
+    while current_node < left or current_node < right:
+      #print(f'{current_node} > {left} and {right}')
+      if left > current_node and left >= right:
+        self.storage[(2 * index) + 1], self.storage[index] = self.storage[index], self.storage[(2*index) + 1]
```

Comment:
For a line of code like this with a lot of math operations, it is good practice to either break it down by defining new variables (e.g. parent_element = self.storage[index]), or add some comments to explain what is going on.